src/lib/handlePageRequest.ts

```diff
@@ -3,91 +3,105 @@ import {
   GetServerSidePropsContext,
   GetServerSidePropsResult,
 } from "next";
-import { NextApiRequestCookies } from "next/dist/next-server/server/api-utils";
 import { FormJSON, FormManager } from "./form/formManager";
 import { CacheEntry } from "./formCache";
 import {
-  getCache,
-  parseFormData,
+  BeaconsContext,
+  decorateGetServerSidePropsContext,
   updateFormCache,
   withCookieRedirect,
 } from "./middleware";
-import { acceptRejectCookieId } from "./types";
+import { Registration } from "./registration/registration";
 
 type TransformCallback = (formData: CacheEntry) => CacheEntry;
 
+export type DestinationIfValidCallback = (context: BeaconsContext) => string;
+
 export type FormManagerFactory = (formData: CacheEntry) => FormManager;
 
 export interface FormPageProps {
   form: FormJSON;
   showCookieBanner?: boolean;
+  flattenedRegistration?: CacheEntry;
 }
 
 export const handlePageRequest = (
   destinationIfValid: string,
```

Comment:
Was in two minds as to whether to replace all the forms with the function argument on line 33.  Half tempted to remove it as we pass in the context that now has additional information on it but would involve more changes on each page.

This isn't needed now as we have the callback but I have left it in for now so that we don't have to update every page that uses this function.